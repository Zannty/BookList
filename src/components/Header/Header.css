.header {
  display: flex;
  height: 60px;
  background-color: rgb(211, 211, 211);
  padding: 0.3em;
  border-bottom: 4px double black;
  font-family: "Oswald", sans-serif;
}
.main-title {
  font-weight: bold;
}
.search {
  display: flex;
  max-width: 100%;
  margin-left: auto;
  letter-spacing: 2px;
}
.input-search {
  padding-right: 10px;
  margin-right: 10px;
  margin-left: 2px;
}
.select-search {
  margin-right: 10px;
  margin-left: 2px;
}
.btn-search {
  height: 2.3em;
}
