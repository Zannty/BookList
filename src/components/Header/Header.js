import React, { useEffect } from "react";
import "./Header.css";
import {
  handleInputChange,
  handleFilterAuthorChange,
  handleFilterGenreChange,
  handleFilterNameChange,
  handleApplyFilter
} from "./../../actions/action";
import { connect } from "react-redux";
export const Header = ({
  handleInputChange,
  filterSearchValue,
  handleFilterAuthorChange,
  handleFilterGenreChange,
  handleFilterNameChange,
  handleApplyFilter
}) => {
  const sub = e => {
    e.preventDefault();
    handleApplyFilter();
  };
  return (
    <div className="header">
      <h2 className="main-title">Список книг</h2>
      <form onSubmit={sub} className="search">
        Критерии поиска
        <p className="checkbox-filter">
          Название
          <input type="checkbox" value="a2" onChange={handleFilterNameChange} />
        </p>
        <p className="checkbox-filter">
          Автор
          <input
            type="checkbox"
            value="a3"
            onChange={handleFilterAuthorChange}
          />
        </p>
        <p className="checkbox-filter">
          Жанр
          <input
            type="checkbox"
            value="a4"
            onChange={handleFilterGenreChange}
          />
        </p>
        Поиск
        <input
          className="form-control form-control-sm input-search"
          type="text"
          onChange={e => handleInputChange(e.target.value)}
          value={filterSearchValue}
        />
        <button
          type="submit"
          onClick={sub}
          className="btn btn-secondary btn-search"
        >
          Применить
        </button>
      </form>
    </div>
  );
};
const mapStateToProps = ({ filterSearchValue }) => {
  return {
    filterSearchValue
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleInputChange: text => dispatch(handleInputChange(text)),
    handleFilterAuthorChange: () => dispatch(handleFilterAuthorChange()),
    handleFilterGenreChange: () => dispatch(handleFilterGenreChange()),
    handleFilterNameChange: () => dispatch(handleFilterNameChange()),
    handleApplyFilter: () => dispatch(handleApplyFilter())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
