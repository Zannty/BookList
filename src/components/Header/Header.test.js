import React from "react";
import { Header } from "./Header";
describe("Header", () => {
  const mockBtn = jest.fn();
  const props = {
    search: "Дефо",
    selectedItem: "author",
    buttonUse: mockBtn
  };
  const container = mount(<Header {...props} />);

  it("snap Header", () => {
    expect(container.debug()).toMatchSnapshot();
  });
  it("props search currect", () => {
    expect(container.props().search).toEqual(props.search);
  });
  it("props select currect", () => {
    expect(container.props().selectedItem).toEqual(props.selectedItem);
  });

  it("buttonUse сlear search", () => {
    container.find("button").simulate("click", {
      preventDefault: () => {}
    });

    expect(mockBtn).toHaveBeenCalledTimes(1);
  });
});
