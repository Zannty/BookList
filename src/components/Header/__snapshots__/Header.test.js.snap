// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snap Header 1`] = `
"<Header search=\\"Дефо\\" selectedItem=\\"author\\" buttonUse={[Function: mockConstructor]}>
  <div className=\\"header\\">
    <h2 className=\\"main-title\\">
      Список книг
    </h2>
    <form onSubmit={[Function: sub]} className=\\"search\\">
      Критерий поиска
      <select value=\\"author\\" onChange={[Function: onChange]} className=\\"custom-select select-search\\">
        <option value=\\"def\\">
          По умолчанию
        </option>
        <option value=\\"name\\">
          По названию
        </option>
        <option value=\\"author\\">
          По автору
        </option>
        <option value=\\"genre\\">
          По жанру
        </option>
      </select>
      Поиск
      <input className=\\"form-control form-control-sm input-search\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"Дефо\\" />
      <button type=\\"submit\\" onClick={[Function: sub]} className=\\"btn btn-secondary btn-search\\">
        Применить
      </button>
    </form>
  </div>
</Header>"
`;
